src/plugins/validation/EcdsaValidation.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.25;
+
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {MessageHashUtils} from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+import {PackedUserOperation} from "@eth-infinitism/account-abstraction/interfaces/PackedUserOperation.sol";
+import {SignatureChecker} from "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+
+import {IPlugin, PluginManifest, PluginMetadata} from "../../interfaces/IPlugin.sol";
+import {IValidation} from "../../interfaces/IValidation.sol";
+import {BasePlugin} from "../BasePlugin.sol";
+import {IEcdsaValidation} from "./IEcdsaValidation.sol";
+
+/// @title ECSDA Validation
+/// @author ERC-6900 Authors
+/// @notice This validation enables any ECDSA (secp256k1 curve) signature validation. It handles installation by
+/// each entity (validationId).
+/// Note: Uninstallation will NOT disable all installed validation entities. None of the functions are installed on
+/// the account. Account states are to be retrieved from this global singleton directly.
+///
+/// - This validation supports ERC-1271. The signature is valid if it is signed by the owner's private key
+/// (if the owner is an EOA) or if it is a valid ERC-1271 signature from the
+/// owner (if the owner is a contract).
+///
+/// - This validation supports composition that other validation can relay on entities in this validation
+/// to validate partially or fully.
+contract EcdsaValidation is IEcdsaValidation, BasePlugin {
```

Comment:
The plan is to delete `SingleOwnerPlugin` after this as it essentially supports a subset  feature of `EcdsaValidation`.
The features of `EcdsaValidation` is in the docs above.

Reply:
`EcdsaValidation` supports contract owners via ERC-1271 too.

Reply:
Ah, sorry I missed that. In that case, isn't Single owner a more applicable name than ECDSA? Since it's not always ECDSA?

Reply:
hmmm, what do you mean? What could be an owner for SingleOwner that cannot be a signer for EcdsaVa?

Reply:
I meant "SingleOwnerPlugin" is possibly a more fitting name. Via 1271, the signature validation is no longer just ECDSA (using `ecrecover`), because the owner contract can implement any type of check.

Reply:
Gotcha. Good point. Maybe `SingleSigner` is a bit more fitting.

Reply:
But then why are we recreating it in a new plugin contract, rather than updating SingleOwnerPlugin, if it's doing the same thing? I think we're one line away from equivalence, with the update to `userOp.sender`

Reply:
They are very similar in the end (each func will need some update). Creating the validation and add tests was a lot easier for the development process than updating at the time.
Also it is a minor repo structure improvement with validation v.s. owner folder.
With [PR [4/n]](https://github.com/erc6900/reference-implementation/pull/97), it does not matter how we get to here, right?